### DISCUSSION TITLE
v2: chainable options API

### DISCUSSION THREAD

**Discussion by moorereason** [pelletier/go-toml#582] [Questions] [answered]
v2 (40cfb6f45824f8f5cd16957894b5c42e10b65f6e) doesn't support chaining of configuration options when building an encoder/decoder.  Was that intentional?

For example, these two blocks accomplish the same thing:

```go
// v1
err := toml.NewEncoder(buf).ArraysWithOneElementPerLine(multi).Encode(v)

// v2
enc := toml.NewEncoder(buf)
enc.SetArraysMultiline(multi)
err := enc.Encode(v)
```

**Comment by pelletier** [comment, marked as answer]
This was not intentional! Adding this to the list of things to do after I figure out this Document feature.

**Reply by pelletier** [reply]
This has been fixed in https://github.com/pelletier/go-toml/pull/670.
